### PR TITLE
1.11 Should Support PHP 7.1

### PR DESCRIPTION
**Pull request by greezybacon**
This is an effort to fix all the warnings and crashes running osTicket on PHP 7.1

### CODE REVIEW

**Comment by JediKev** [comment]
🙏🏻🙏🏻🙏🏻

File: include/class.forms.php [hunk 1]
```diff
@@ -4364,7 +4364,7 @@ class FileUploadWidget extends Widget {
         ),
     );
 
-    function render($options) {
```
Comment (SalvatoreAzad) [inline include/class.forms.php]: I think this is line 4376

**Comment by husseinsheikh** [comment]
where is the download button for this pull?

**Comment by ntozier** [comment]
add .patch to the end of the url for a patch file.  
Or .diff to the end of the url for a diff file. 

If you are not familiar with .patch or .diff files then I recommend reading this:
https://www.thegeekstuff.com/2014/12/patch-command-examples
If this doesn't make sense there is a lot of examples on Google.

Or lastly you can click on the files changed tab and make the changes manually.

**Comment by Ketrel** [comment]
Is this still needed for the develop-next branch as of the 2fd4a42 commit?

**Comment by JediKev** [comment]
@Ketrel

It has not been merged so ya.

Cheers.

**Comment by Ketrel** [comment]
@JediKev 

Just wanted to make sure since it no longer merges clean.  I was looking through the diff and it looks like some of the blocks it would remove have changed entirely or have new lines, so I wasn't sure if it was still applicable at all.

**Comment by JediKev** [comment]
@GrahamKendall 

Nope, for that you'll have to view each file in the Files tab, and download each one individually. (Github doesn't give you a download of the full pull request, mostly likely because it's way faster and easier to apply a patch or to just pull the changes to your own clone/fork)

The easiest way to get these changes is to tack `.patch` to the end of the pull URL (https://github.com/osTicket/osTicket/pull/4344.patch), get a patch file, and apply that patch to your clone/fork of the repo.

Cheers.

**Comment by t-oster** [comment]
Well @greezybacon @protich , what is the best way to help? Should I send a Pull request based on this pull request or how to proceed? I have the feeling, that you are spammed with Issues and Pull requests, so I want to help in the most efficient way.
I am planning to fix all PHP7.1+ issues and then add a few signals/hooks for better plugin support.

**Comment by GrahamKendall** [comment]
Hi All, yesterday morning I deployed 1.11 after commit 13f2f4a and that seems to have fixed out New Message Alerting.

We did have to rebuild the criteria for a custom queue page.

It also seems we have lost the "save" button from editing existing organisation notes, but I'll post that separately.

Very Many Thanks,
Graham

**Pull request merged**